src/bitops.c

```diff
@@ -697,59 +894,87 @@ void bitopCommand(client *c) {
     /* Compute the bit operation, if at least one string is not empty. */
     if (maxlen) {
         res = (unsigned char*) sdsnewlen(NULL,maxlen);
-        unsigned char output, byte;
+        unsigned char output, byte, disjunction, common_bits;
         unsigned long i;
+        int useAVX2 = 0;
+
+        /* Number of bytes processed from each source key */
+        j = 0;
+
+#if defined(HAVE_AVX2)
+        if (BITOP_USE_AVX2) {
+            j = bitopCommandAVX(src, res, op, numkeys, minlen);
 
-        /* Fast path: as far as we have data for all the input bitmaps we
+            serverAssert(minlen >= j);
+            minlen -= j;
+
+            useAVX2 = 1;
+        }
+#endif
+
+#if !defined(USE_ALIGNED_ACCESS)
+        /* We don't have AVX2 but we still have fast path:
+         * as far as we have data for all the input bitmaps we
          * can take a fast path that performs much better than the
          * vanilla algorithm. On ARM we skip the fast path since it will
          * result in GCC compiling the code using multiple-words load/store
          * operations that are not supported even in ARM >= v6. */
-        j = 0;
-        #ifndef USE_ALIGNED_ACCESS
-        if (minlen >= sizeof(unsigned long)*4 && numkeys <= 16) {
-            unsigned long *lp[16];
+        if (!useAVX2 && minlen >= sizeof(unsigned long)*4) {
```

Comment:
Whether this is readable will be higher. It ensures that when entering fastpath, we should not have already entered bitopCommandAVX().
and we can add some comments for this assertion.

```suggestion
        if (minlen >= sizeof(unsigned long)*4) {
            serverAssert(!useAVX2);
```